framework/framework.go

```diff
@@ -197,6 +203,16 @@ func (f *Framework) UpdateFunc(oldObj, newObj interface{}) {
 	f.mutex.Lock()
 	defer f.mutex.Unlock()
 
+	ok, err := f.addFinalizer(obj)
+	if err != nil {
+		f.logger.Log("event", "update", "function", "UpdateFunc", "level", "error", "message", "stop framework reconciliation due to error", "stack", fmt.Sprintf("%#v", err))
+		return
+	}
+	if ok {
+		f.logger.Log("event", "update", "function", "UpdateFunc", "level", "debug", "message", "stop framework reconciliation, finalizer added")
+		return
```

Comment:
Does that mean when the object is reconciled once it won't reconcile anymore? That isn't correct. What happens when a resource degrades because of external factor? This also prevents reconciling on the object update.

Reply:
No, we return `false` if there was already a finalizer - so reconciliation will continue as long as there is a finalizer:
https://github.com/giantswarm/operatorkit/blob/08fef601dfba839fe981b741de6770342470f946/framework/finalizer.go#L30

Reply:
Ack. Makes total sense.